### PR TITLE
S2 add item

### PR DESCRIPTION
**Pull request by nesclark**

### CODE REVIEW

File: addplant.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+session_start();
```
Comment (charliecog) [inline addplant.php]: unsused (all the php on this page) so can remove

File: functions.php [hunk 2]
```diff
@@ -54,5 +54,50 @@ function callPlant(array $plants): string
...
+ */
+
+function validateNewPlant(string $colloquial_name_input, string $latin_name_input, string $size_input, string $image_input): bool
+{
+    $colloquial_name_input = filter_var($colloquial_name_input, FILTER_SANITIZE_STRING);
+    $latin_name_input = filter_var($latin_name_input, FILTER_SANITIZE_STRING);
+    $size_input = filter_var($size_input, FILTER_SANITIZE_STRING);
+    $image_input = filter_var($image_input, FILTER_SANITIZE_URL);
 
+    $colloquial_name_input = preg_match('/^[a-zA-Z\s]+$/', $colloquial_name_input);
```
Comment (charliecog) [inline functions.php]: this regex is allowing through name entries of space bars eg: '          '

File: functions.php [hunk 3]
```diff
@@ -54,5 +54,50 @@ function callPlant(array $plants): string
...
+    $latin_name_input = preg_match('/^[a-zA-Z\s]+$/', $latin_name_input);
```
Comment (charliecog) [inline functions.php]: this regex is allowing through name entries of space bars eg: ' '

File: index.php [hunk 4]
```diff
@@ -1,4 +1,7 @@
 <?php
+
+session_start();
```
Comment (charliecog) [inline index.php]: not used, can remove

File: invalidpage.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
+
+
+
```
Comment (charliecog) [inline invalidpage.php]: extra line breaks can be removed

File: invalidpage.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    <ul>
+        <li>
+            <a href="index.php">Home</a>
+        </li>
+    </ul>
+</nav>
+<header>
+    <div class="input-container">
+        <h3>Invalid entry</h3>
+        <form action="addplant.php">
```
Comment (charliecog) [inline invalidpage.php]: this could just be an a tag

File: postdata.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+$colloquial_name_input = $_POST['colloquial_name'];
+$latin_name_input = $_POST['latin_name'];
+$size_input = $_POST['size_cm'];
+$image_input = $_POST['image_url'];
+
+$input_value = validateNewPlant($colloquial_name_input, $latin_name_input, $size_input, $image_input);
+
+$db = createDbConnection();
```
Comment (charliecog) [inline postdata.php]: move this inside the if statement so it will only be spun up if needed

**Review by charliecog — COMMENTED** [review]
looks good - almost there!

File: addplant.php (hunk 1 above)
Comment (nesclark) [reply addplant.php]: removed

File: index.php (hunk 4 above)
Comment (nesclark) [reply index.php]: removed

File: invalidpage.php (hunk 5 above)
Comment (nesclark) [reply invalidpage.php]: removed

File: postdata.php (hunk 7 above)
Comment (nesclark) [reply postdata.php]: done

File: invalidpage.php (hunk 6 above)
Comment (nesclark) [reply invalidpage.php]: done

File: functions.php (hunk 2 above)
Comment (nesclark) [reply functions.php]: fixed

File: functions.php (hunk 3 above)
Comment (nesclark) [reply functions.php]: fixed

File: postdata.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+require_once 'functions.php';
+
+session_start();
```
Comment (charliecog) [inline postdata.php]: can remove the sesh start as not being used

File: postdata.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+$colloquial_name_input = $_POST['colloquial_name'];
+$latin_name_input = $_POST['latin_name'];
+$size_input = $_POST['size_cm'];
+$image_input = $_POST['image_url'];
+
+$input_value = validateNewPlant($colloquial_name_input, $latin_name_input, $size_input, $image_input);
+
+
+if ($input_value === true) {
```
Comment (charliecog) [inline postdata.php]: could just make this if($input) because of the rules of truthy/falsey

**Review by charliecog — COMMENTED** [review]
so close! last couple tweaks

**Review by charliecog — APPROVED** [review]

**Pull request merged**